src/Reference_Many.php

```diff
@@ -86,22 +86,46 @@ public function refLink($defaults = [])
      */
     public function addField($n, $defaults = [])
     {
-        if (!isset($defaults['aggregate'])) {
+        if (!isset($defaults['aggregate']) && !isset($defaults['expr'])) {
             throw new Exception([
-                '"aggregate" strategy should be defined for oneToMany field',
+                '"aggregate" strategy (or "expr") should be defined for oneToMany field',
                 'field'    => $n,
                 'defaults' => $defaults,
             ]);
         }
 
-        $field = isset($defaults['field']) ? $defaults['field'] : $n;
+        $field_n = isset($defaults['field']) ? $defaults['field'] : $n;
+        $field = isset($defaults['field']) ? $defaults['field'] : null;
 
-        $e = $this->owner->addExpression($n, array_merge([
-            function () use ($defaults, $field) {
-                return $this->refLink()->action('fx0', [$defaults['aggregate'], $field]);
-            }, ],
-            $defaults
-        ));
+        if (isset($defaults['expr'])) {
+            $cb = function () use ($defaults, $field) {
+                $r = $this->refLink();
+
+                return $r->action('field', [$r->expr(
+                    $defaults['expr'],
+                    isset($defaults['args']) ? $defaults['args'] : null
+                ), 'alias'=>$field]);
+            };
+            unset($defaults['args']);
+        } elseif (is_object($defaults['aggregate'])) {
+            $cb = function () use ($defaults, $field) {
+                return $this->refLink()->action('field', [$defaults['aggregate'], 'alias'=>$field]);
+            };
+        } elseif ($defaults['aggregate'] == 'count' && !isset($defaults['field'])) {
+            $cb = function () use ($defaults, $field) {
+                return $this->refLink()->action('count', ['alias'=>$field]);
```

Comment:
I don't see where it sets `'field' => new Expression('*')` here.
Also no test coverage for this case (fixed).
And it doesn't use `fx0` as it does few lines below. Maybe that's not needed in case of `count`, because I think in SQL count(whatever) always returns number no matter if it's 0 or it selects `*` or some nullable field.

Reply:
action('count') does that. i think it was in the test.. coverage broken?

Reply:
ok